src/main/java/freiman/conway/GridComponent.java

```diff
@@ -0,0 +1,66 @@
+package freiman.conway;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+
+public class GridComponent extends JComponent {
+    private Grid grid;
+    private final int gridSpacing = 15;
+
+
+    public GridComponent(Grid grid) {
+        this.grid = grid;
+        addMouseListener(new MouseListener() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                if (!grid.isAlive(e.getX() / gridSpacing, e.getY() / gridSpacing)) {
+                    grid.put(e.getX() / gridSpacing, e.getY() / gridSpacing);
+                } else {
+                    grid.remove(e.getX() / gridSpacing, e.getY() / gridSpacing);
+                }
```

Comment:
call getX and getY and do the division only once and save the value. Makes this more readable.